### PR TITLE
feat(schema): add methods for proto <> schema conversions

### PR DESCRIPTION
**Pull request by raulb**
### Description

This will be used on https://github.com/ConduitIO/conduit-connector-protocol/pull/96

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-commons/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Review by lovromazgon — COMMENTED** [review]
Btw, seems like we need to `go mod tidy` to make the CI action happy.

File: schema/proto.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+// -- From Schema To Proto ----------------------------------------------------
+
+// ToProto takes data from the receiver and populates the supplied proto object.
+// If the function returns an error, the proto object could be partially
+// populated.
+func (s *Schema) ToProto(proto *schemav1.Schema) error {
+	if proto == nil {
+		return nil
```
[the remark this answers is not in the document]
Comment (raulb) [reply schema/proto.go]: @lovromazgon let me know what you think https://github.com/ConduitIO/conduit-commons/pull/72/commits/ef0d77529f8efd3a63443afffbd64572f8c544ec

File: go.mod [hunk 2]
```diff
@@ -60,7 +60,7 @@ require (
 	github.com/butuzov/mirror v1.2.0 // indirect
 	github.com/catenacyber/perfsprint v0.7.1 // indirect
 	github.com/ccojocar/zxcvbn-go v1.0.2 // indirect
-	github.com/cespare/xxhash/v2 v2.1.2 // indirect
+	github.com/cespare/xxhash/v2 v2.2.0 // indirect
```
Comment (lovromazgon) [inline go.mod:63]: btw, is there a reason we're updating an indirect dependency here?

File: schema/errors.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import "errors"
+
+// errInvalidProtoIsNil is returned when trying to convert a proto schema to a schema object and the proto is nil.
```
Comment (lovromazgon) [inline schema/errors.go:19]: Actually it's the other way around, `ToProto` converts from schema to proto object.

Comment (raulb) [reply schema/errors.go]: @lovromazgon good catch 🤦

**Review by lovromazgon — APPROVED** [review]
🚀

**Pull request merged**